### PR TITLE
#130 - HMDA upload

### PR DESCRIPTION
**Pull request by emkap01**
I think/hope I was able to roll the marble end-to-end and add 250MB of Home Mortgage Disclosure Act records to the project, per issue #130.  

I followed all steps in the [how-to guide](http://housinginsights.org/resources/adding-data-sources.html) to get the records cleaned and added to the Postgres db, as well as push a copy of the raw records to the S3 bucket.   I did *_not_* attempt to join the new table to any other tables in the db, although I _did_ add a function to my cleaner class to make sure that the value we can use for that join (census_tract_number) is consistent with what's already in the db.  I will play around with the joins next.

This is the first pull request I've submitted on this project, and I'm very much an amateur at Python, so please don't hesitate to let me know if I messed something up.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -10,14 +10,7 @@ services:
             - ./docs:/srv/jekyll
         ports:
             - "4000:4000"
-    web:
```
Comment (NealHumphrey) [inline docker-compose.yml]: @emkap01 can you remove this file from the pull request? We modified this version to get it to work locally for you, but don't want that in everyone's version. Here's instructions: http://stackoverflow.com/questions/12481639/remove-files-from-git-commit

If you're not comfortable doing this, we can do it in person on Tuesday instead.

File: python/scripts/meta.json [hunk 2]
```diff
@@ -2083,6 +2083,558 @@
...
+                "source_name": "application_date_indicator",
+                "sql_name": "application_date_indicator",
+                "type": "text"
+            },
+            {
+                "display_name": "as_of_year",
+                "display_text": "",
+                "source_name": "as_of_year",
+                "sql_name": "as_of_year",
+                "type": "text"
```
Comment (NealHumphrey) [inline python/scripts/meta.json]: should this be a date? Or an integer?

File: python/scripts/meta.json [hunk 3]
```diff
@@ -2083,6 +2083,558 @@
...
+                "source_name": "loan_amount_000s",
+                "sql_name": "loan_amount_000s",
+                "type": "integer"
+            },
+            {
+                "display_name": "number_of_1_to_4_family_units",
+                "display_text": "",
+                "source_name": "number_of_1_to_4_family_units",
+                "sql_name": "number_of_1_to_4_family_units",
+                "type": "decimal"
```
Comment (NealHumphrey) [inline python/scripts/meta.json]: should this be integer?

File: python/housinginsights/ingestion/Cleaners.py [hunk 4]
```diff
@@ -308,6 +308,21 @@ def replace_tracts(self,row,row_num,column_name='census_tract'):
             logging.warning('  no matching Tract found for row {}'.format(row_num,row))
         return row
 
+    def append_tract_label(self,row,row_num,column_name='census_tract_number'):
```
Comment (NealHumphrey) [inline python/housinginsights/ingestion/Cleaners.py]: This looks great!

**Comment by NealHumphrey** [comment]
@emkap01 I was able to get this to run on my machine and successfully imported the data - good stuff! Just a couple minor edits and this will be good to go.

File: docker-compose.yml (hunk 1 above)
Comment (emkap01) [inline docker-compose.yml]: I took a shot at updating the pull request in order to remove docker-compose.yml via the instructions in that link, let me know if it didn't work.

File: python/scripts/meta.json (hunk 2 above)
Comment (emkap01) [inline python/scripts/meta.json]: The values in this field will consistently be a 4-digit year.  I wasn't sure if they should be dates or not, so I cast them as text.

File: python/scripts/meta.json (hunk 3 above)
Comment (emkap01) [inline python/scripts/meta.json]: No, I examined a subset of these values as a spreadsheet and some of them are actually decimals.

File: python/housinginsights/ingestion/Cleaners.py (hunk 4 above)
Comment (emkap01) [inline python/housinginsights/ingestion/Cleaners.py]: Thanks!  This was about as far as I could stretch my amateur Python skills, but I was really excited to discover that it worked when I looked at the records in my local copy of the db.

File: python/scripts/meta.json (hunk 2 above)
Comment (NealHumphrey) [inline python/scripts/meta.json]: Let's make this an integer then - that way we can do comparison (>, < ) etc. on the years if needed.

**Comment by NealHumphrey** [comment]
@emkap01 it doesn't look like any new commits have been added - have you pushed your upates yet?

**Comment by emkap01** [comment]
Apparently I screwed something up - it looks like I overwrote the files in my pull request when I was trying to remove the docker-compose.yml file, because the versions I included in the pull request no longer exist in my local or remote repos.  If you have the ability to resolve the issue from your end, go ahead, otherwise we can just fix it on Tuesday night.

**Comment by NealHumphrey** [comment]
@emkap01 I can use the copy I pulled to fix it - will do that before Tuesday's session and we can make sure it's all squared away then.

**Comment by terrysky18** [comment]
I've got the docker-compose.yml file on my repository and local drive.  Let me know if you need it.

**Comment by emkap01** [comment]
As discussed I'm just going to close the pull request, delete the branch, then create a new branch with the correct files and submit a new pull request based on that.

**Pull request closed without merging**